src/test/regress/sql/subselect.sql

```diff
@@ -517,3 +517,44 @@ fetch backward all in c1;
 
 commit;
 --end_ignore
+
+-- Ensure that ORCA correctly builts Query dxl tree and all attributes inside
+-- the nested SubLink, which are referenced in GROUP BY clause, are processed
+-- during query normalization. The inner part of SubPLan should contain only t.j
+-- start_ignore
+DROP TABLE IF EXISTS t;
+DROP TABLE IF EXISTS s;
+-- end_ignore
+CREATE TABLE t (i int, j int) distributed by (i);
+CREATE TABLE s (i int, j int) distributed by (i);
+INSERT INTO t VALUES (1, 2);
+explain (verbose, costs off)
+SELECT j,
+(SELECT j FROM (SELECT j) q2)
+FROM t GROUP BY i, j;
+SELECT j,
```

Comment:
Add please an empty line or make it easier to read the queries in another way

Reply:
fixed